modules/spo-replicating-controllers.adoc

```diff
@@ -0,0 +1,131 @@
+// Module included in the following assemblies:
+//
+// * security/security_profiles_operator/spo-seccomp.adoc
+// * security/security_profiles_operator/spo-selinux.adoc
+
+ifeval::["{context}" == "spo-seccomp"]
+:seccomp:
+:type: seccomp
+:kind: SeccompProfile
+endif::[]
+ifeval::["{context}" == "spo-selinux"]
+:selinux:
+:type: SELinux
+:kind: SelinuxProfile
+endif::[]
+
+:_content-type: PROCEDURE
+[id="spo-replicating-controllers_{context}"]
+= Replicating controllers and SecurityContextConstraints
+
+When deploying {type} policies for replicating controllers, such as deployments or daemonsets, note that the `Pod` objects spawned by the controllers are not running with the identity of the user who creates the workload. Unless a `ServiceAccount` is selected, the pods might revert to using a restricted `SecurityContextConstraints` (SCC) which does not allow use of custom security policies.
+
+.Procedure
+
+. Create the following SCC object to allow SELinux policies to be used in the `nginx-secure` namespace:
++
+[source,yaml]
+----
+apiVersion: security.openshift.io/v1
+kind: SecurityContextConstraints
+metadata:
+  annotations:
+    kubernetes.io/description: A special SCC for running nginx with a custom SELinux policy
+  name: nginx-secure
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPID: false
+allowHostPorts: false
+allowPrivilegeEscalation: true
+allowPrivilegedContainer: false
+allowedCapabilities: null
+defaultAddCapabilities: null
+fsGroup:
+  type: MustRunAs
+priority: null
+readOnlyRootFilesystem: false
+requiredDropCapabilities:
+- KILL
+- MKNOD
+- SETUID
+- SETGID
+runAsUser:
+  type: MustRunAsRange
+seLinuxContext:
+  type: MustRunAs
+  seLinuxOptions:
+    type: test-selinux-recording-nginx-0_nginx-secure.process
```

Comment:
I didn't quite clear why the scc with selinuxprofiles needed when trying to apply a seccompprofile to a deployment. @jhrozek Could  you help to explain? Thanks.

Reply:
(I haven't read the whole example, just replying to this comment specifically)

When a pod is not created directly by the user, but by a replicating controller (Deployment, DeamonSet, Job, ..), then the pod is created by the identity that controller is running as, so typically by the SA that the controller is used. And by default, that SA would be matched against one of the restrictive SCCs which don't allow you to set custom SELinux context. So the only way to make this work is to create an SA and add a roleBinding to a role that uses a more permissive SCC (with the privileged one being the most permissive).

I'll read the whole example later..

Reply:
upstream PR opened: https://github.com/kubernetes-sigs/security-profiles-operator/pull/1318